node/node.go

```diff
@@ -249,7 +252,12 @@ const (
 // node status update (because some things still expect the node to be updated
 // periodically), otherwise it will only use node status update with the configured
 // ping interval.
-func (n *NodeController) Run(ctx context.Context) error {
+func (n *NodeController) Run(ctx context.Context) (retErr error) {
+	defer func() {
+		n.err = retErr
```

Comment:
I believe this can trigger the go race detector, as the race detector doesn't treat (closing) channels as synchronization points.

Reply:
Indeed it does. Fixed to use a mutex.